src/quilla/plugins/base_storage.py

```diff
@@ -0,0 +1,188 @@
+            bytes object if the image is not found.
+        '''
+
+    @abstractmethod
+    def cleanup_reports(self):
+        '''
+        Searches for reports that match some cleanup criteria
+        '''
+
+    @abstractmethod
+    def store_treatment_image(
+        self,
+        run_id: str,
+        baseline_id: str,
+        treatment: bytes,
+    ) -> str:
+        '''
+        Stores a treatment image within the storage mechanism enabled by the plugin
+
+        Args:
+            run_id: The run ID of the current Quilla run, to version the treatment images
+            baseline_id: The ID of the baseline that this treatment image is associated
+                with
+            treatment: The image data in bytes
+
+        Returns:
+            An identifier that can locate the newly stored treatment image
+        '''
+
+    @abstractmethod
+    def store_baseline_image(
+        self,
+        baseline_id: str,
+        baseline: bytes,
+    ) -> str:
+        '''
+        Stores a baseline image under the given baseline_id.
+
+        This function should be used to update the current baseline
+        image, or create a new one if the baseline did not previously exist
+
+        Args:
+            baseline_id: A unique identifier for the image
+            baseline: The image data in bytes
+
+        Returns:
+            A URI for the new baseline image
+        '''
+
+    @abstractmethod
+    def make_baseline_uri(
+        self,
+        run_id: str,
+        baseline_id: str
+    ) -> str:
+        '''
+        Generates a baseline URI for the current run given the baseline_id of the image.
+
+        It is recommended that plugins create a clone of the baseline image
+        when generating the URI so that the returned URI will uniquely identify
+        the baseline that was used for the associated run ID. This ensures that
+        even if the baseline image is updated, the report is still meaningful.
+
+        Args:
+            run_id: The unique ID identifying the current run
+            baseline_id: The unique identifier for the image
+
+        Returns:
+            A URI that can locate the baseline image used for the given run
+        '''
+
+    def get_image(self, baseline_id: str) -> Optional[bytes]:
+        '''
+        Determines if the plugin should run, and if so searches for the image
+        with the specified baseline ID and returns the byte data for it
+
+        Args:
+            baseline_id: The unique ID to search for
+
+        Returns:
+            None if the plugin is not enabled, a ``bytes`` representation
+            of the image if it is. If no baseline image is found, returns an
+            empty byte string
+        '''
+        if not self.is_enabled:
+            return None
+
+        return self.find_image_by_baseline(baseline_id)
+
+    def get_baseline_uri(self, run_id: str, baseline_id: str) -> Optional[str]:
```

Comment:
any chance we can combine make_baseline_uri() and get_baseline_uri()?

Reply:
Same answer from above- this is used as the method that should be called from the plug-in hook, to ensure that simple mistakes like forgetting to check for is_enabled doesn’t mess up quilla